### PR TITLE
add cross-entropy loss validation metric in the binary classifier case

### PR DESCRIPTION
**Pull request by davetroiano**
@kexinhuang12345 I added validation logistic loss to the binary case for some comparison testing vs the Determined DeepPurpose port; figured it would be a nice to have in the DP codebase

### CODE REVIEW

**Comment by kexinhuang12345** [comment]
Hey, @davetroiano thanks for the pull request! It's indeed a useful feature. 

Have you tried to run the four case studies in the DEMO? I think it is likely we may use test_ function in other files such as the one-liner mode. Would love to see if this does not break them. Thanks!

**Comment by davetroiano** [comment]
> Hey, @davetroiano thanks for the pull request! It's indeed a useful feature.
> 
> Have you tried to run the four case studies in the DEMO? I think it is likely we may use test_ function in other files such as the one-liner mode. Would love to see if this does not break them. Thanks!

@kexinhuang12345 I ran Case Study 1 (a), Case Study 1 (b), and Case Study 2 on the repo [README](https://github.com/kexinhuang12345/DeepPurpose) and those are working.  And then this is the snippet I was using to test:

```
import DeepPurpose.models as models
from DeepPurpose.utils import *
from DeepPurpose.dataset import *
X_drug, X_target, y = load_process_DAVIS('/data/DAVIS', binary=True)
drug_encoding = 'CNN'
target_encoding = 'CNN'
train, val, test = data_process(X_drug, X_target, y, 
                                drug_encoding, target_encoding, 
                                split_method='random',frac=[0.7,0.1,0.2])
config = generate_config(drug_encoding = drug_encoding, 
                         target_encoding = target_encoding, 
                         cls_hidden_dims = [1024,1024,512], 
                         train_epoch = 1000, 
                         LR = 0.001, 
                         batch_size = 256,
                         cnn_drug_filters = [32,64,96],
                         cnn_target_filters = [32,64,96],
                         cnn_drug_kernels = [4,6,8],
                         cnn_target_kernels = [4,8,12]
                        )
model = models.model_initialize(**config)
model.train(train, val, test)
```

Are there other usage patterns to test or does this cover the case studies?

**Comment by kexinhuang12345** [comment]
Thanks! Should be good!
